sdk/core/azure-core/azure/core/credentials.py

```diff
@@ -2,16 +2,29 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License. See LICENSE.txt in the project root for
 # license information.
-# --------------------------------------------------------------------------
-from typing_extensions import Protocol
+# -------------------------------------------------------------------------
+from typing import TYPE_CHECKING
+
+
+if TYPE_CHECKING:
+    from typing import NamedTuple
+    from typing_extensions import Protocol
+
+    AccessToken = NamedTuple("AccessToken", [("token", str), ("expires_on", int)])
+
+    class TokenCredential(Protocol):
```

Comment:
Why is the class definition conditional on type checking?

Reply:
When type checking I want `typing.NamedTuple` so the fields are typed. At runtime I don't know that class will be available (although if I recall correctly it's a transitive dependency of azure-core) and it won't help anyway, so I use `collections.namedtuple` instead.